tests/e2e/test_couchbase.py

```diff
@@ -0,0 +1,328 @@
+                if self.is_up_and_running(couchbase_pod[0], self.up_check):
+                    break
+            except IndexError:
+                log.info("Couchbase operator is not up")
+        cb_work = templating.load_yaml(constants.COUCHBASE_WORKER_SECRET)
+        self.cb_worker = OCS(**cb_work)
+        self.cb_worker.create()
+        cb_example = templating.load_yaml(constants.COUCHBASE_WORKER_EXAMPLE)
+        self.cb_examples = OCS(**cb_example)
+        self.cb_examples.create()
+        # Wait for last of three workers to be running.
+        for cb_wrk_pod in TimeoutSampler(
+            300,
+            3,
+            get_pod_name_by_pattern,
+            'cb-example-0002',
+            self.COUCHBASE_OPERATOR
+        ):
+            try:
+                if self.is_up_and_running(cb_wrk_pod[0], self.up_check):
+                    # once last pod is up, make sure all are ready
+                    counter = 0
+                    for wpodn in range(0, 3):
+                        cbw_pod = f"cb-example-{wpodn:04}"
+                        if self.is_up_and_running(cbw_pod, self.up_check):
+                            counter += 1
+                    if counter == 3:
+                        break
+            except IndexError:
+                log.info("Couchbase workers are not up")
+        pfight = templating.load_yaml(constants.COUCHBASE_BASIC_PILLOWFIGHT)
+        pillowfight = OCS(**pfight)
+        pillowfight.create()
+        pf_completion_info = ''
+        pf_pod = ''
+        for pillowfight_pod in TimeoutSampler(
+            300,
+            3,
+            get_pod_name_by_pattern,
+            'pillowfight',
+            self.COUCHBASE_OPERATOR
+        ):
+            try:
+                pf_pod = pillowfight_pod[0]
+                pod_info = self.up_check.exec_oc_cmd(f"get pods {pf_pod} -o json")
+                pf_status = pod_info['status']['containerStatuses'][0]['state']
+                if 'terminated' in pf_status:
+                    pf_completion_info = pf_status['terminated']['reason']
+                    break
+            except IndexError:
+                log.info("Pillowfight not yet completed")
+        if pf_completion_info == 'Error':
+            raise Exception(
+                "Basic pillowfight failed to complete"
+            )
+        if pf_completion_info == 'Completed':
+            stats = self.parse_couchbase_log(pf_pod)
+            stat1 = min(stats['opspersec'])
+            if stat1 < self.MIN_ACCEPTABLE_OPS_PER_SEC:
+                raise Exception(
+                    f"Worst OPS/SEC value reported is {stat1}"
+                )
+            stat2 = max(stats['resptimes'].keys()) / 1000
+            if stat2 > self.MAX_ACCEPTABLE_RESPONSE_TIME:
+                raise Exception(
+                    f"Worst response time reported is {stat2} milliseconds"
+                )
+            return
+        raise Exception("Invalid return from basic pillowfight")
+
```

Comment:
How do you handle test result?  What is the outcome you expect when test case is completed?

Reply:
A couple of sanity checks are performed on the logs of the pillowfight.  If these pass, then the test should complete successfully.